src/Control/Distributed/Process/Node.hs

```diff
@@ -319,11 +322,38 @@ startServiceProcesses node = do
 
 -- | Force-close a local node
 --
--- TODO: for now we just close the associated endpoint
+-- This function first try to brutally kill all processes, this
+-- process is not guaranteed to continue. So this is application
+-- responsibility to either close all it's processes before dying
+-- or being able to gracefully shutdown.
 closeLocalNode :: LocalNode -> IO ()
-closeLocalNode node =
-  -- TODO: close all our processes, surely!?
-  NT.closeEndPoint (localEndPoint node)
+closeLocalNode node = join $ terminateProcesses
+  where
+    terminateProcesses :: IO (IO ())
```

Comment:
In a new version it doesn't, because this function doesn't start new iteration unless all processes from current generation were killed. this means that for each process there will be only one exception. and then when all processes were killed then there will be a new check if there are new processes. If so then termination procedure continues. There are few other possible solutions:
1. Invent new procedure that will not allow staring new processes when node is closing
2. Use agent as agent has information about processes born and death.